src/state/swap/reducer.ts

```diff
@@ -30,14 +31,17 @@ export interface SwapState {
   readonly feeConfig: FeeConfig | undefined
 }
 
+const { search } = window.location
+const { inputCurrency, outputCurrency } = parse(search.indexOf('?') === 0 ? search.substring(1) : search)
+
 const initialState: SwapState = {
   independentField: Field.INPUT,
   typedValue: '',
   [Field.INPUT]: {
-    currencyId: '',
+    currencyId: inputCurrency?.toString() || '',
   },
   [Field.OUTPUT]: {
-    currencyId: '',
+    currencyId: outputCurrency?.toString() || '',
```

Comment:
if when we visit web the first time if inputCurrency/outputCurrency exist in url. set it to default value of initialState . This will reduce the number of times re-render. 
Because if url is. /swap?inputCurrency=xxx.   => state change from `''` => `xxx`. it cause my app rerender.